src/menu/v1beta1/local-v1.js

```diff
@@ -151,7 +151,40 @@ goog.scope(function() {
      *         the underlying RPC, or during transmission.
      */
     product(key, callback, config) {
+      const that = this;
       return new Promise((resolve, reject) => {
+        /**
+         * Store a local object, after having fetched it from remote services.
+         *
+         * @param {?proto.bloombox.services.menu.v1beta1.GetProduct.Response} r
+         *        RPC response promise.
+         * @private
+         */
+        function storeLocal_(r) {
+
+          // @TODO: store the object locally, fill in keys-only data
+
+          if (callback) callback(r, null);
+          resolve(r);
+        }
+
+        /**
+         * Fetch a product from remote services.
+         *
+         * @private
+         */
+        function fetchRemote_() {
+          // not found
+          that.remote.product(key, function(response, err) {
```

Comment:
this stuff in here is wiring to allow the JS SDK to store menu data locally, then sync it, in IndexedDB